codechain/primitives/PlatformAddress.py

```diff
@@ -0,0 +1,114 @@
+                f"Invalid public key for creating PlatformAddress: {public_key}"
+            )
+
+        return cls.from_account_id(
+            get_account_id_from_public(str(public_key)), **kwargs
+        )
+
+    @classmethod
+    def from_account_id(cls, account_id, **kwargs):
+        if "network_id" in kwargs:
+            network_id = kwargs.get("network_id")
+        else:
+            network_id = "cc"
+        if "version" in kwargs:
+            version = kwargs.get("version")
+        else:
+            version = 1
+
+        if not H160.check(account_id):
+            raise ValueError(
+                f"Invalid account_id for creating PlatformAddress: {account_id}"
+            )
+        if version != 1:
+            raise ValueError(f"Unsupported version for PlatformAddress: {version}")
+        if not isinstance(network_id, str) or len(network_id) != 2:
+            raise ValueError(
+                f"Unsupported network_id for PlatformAddress: {network_id}"
+            )
+
+        return cls(
+            account_id, bech32_encode(network_id + "c", bytes([version]) + account_id)
+        )
+
+    @classmethod
+    def from_string(cls, address: str):
+        if not isinstance(address, str):
+            raise ValueError(f"Expected PlatformAddress string but found: {address}")
+        elif address[2] != "c":
+            raise ValueError(f"Unknown prefix for PlatformAddress: {address}")
+
+        byte = bech32_decode(address[:3], address)
+        version = byte[0]
+
+        if version != 1:
+            raise ValueError(f"Unsupported version for PlatformAddress: {version}")
+
+        account_id = H160(byte[1:])
+
+        return cls(account_id, address)
+
+    @classmethod
+    def check(cls, address):
+        return True if isinstance(address, cls) else cls.check_string(address)
+
+    @classmethod
+    def ensure(cls, address):
+        if isinstance(address, cls):
+            return address
+        elif isinstance(address, str):
+            return cls.from_string(address)
+        else:
+            raise ValueError(
+                f"Expected either PlatformAddress or string but found {address}"
+            )
+
+    @classmethod
+    def ensure_account(cls, address):
+        if isinstance(address, cls):
+            return address.get_account_id()
+        elif isinstance(address, H160):
+            return address
+        elif re.match("^(0x)?[a-fA-F0-9]{40}$", address) is not None:
+            return H160(address)
+        else:
+            return cls.from_string(address).get_account_id()
+
+    @classmethod
+    def check_string(cls, value: str):
+        return None is not re.match(
+            "^.{2}c[qpzry9x8gf2tvdw0s3jn54khce6mua7l]{40}$", value
+        )
+
+    def to_string(self):
+        return self.value
+
+    def get_account_id(self):
+        return self.account_id
+
+
+def get_account_id_from_public(public_key: str):
+    if not isinstance(public_key, str):
+        raise ValueError(
```

Comment:
It should raise `TypeError`. Also `blake160` can accept both `str` and `bytelike`. Change `public_key` type to `Union[str, H512, bytes]` or leave blake160 to handle type error for now. Also you've used this method in ```PlatformAddress.from_public``` which involves unnecessary converting of `bytelike` into `str`

Reply:
I removed unnecessary checking routines.